srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java

```diff
@@ -88,9 +88,11 @@ public void beforeDelete(CdsDeleteEventContext context) {
 
 	private void auditCfgChange(final Action action, final ConfigChange cfgChange, EventContext context) {
 		// create new request context and set tenant to null
-		context.getCdsRuntime().requestContext().modifyUser(user -> user.setTenant(null)).run(ctx -> {
-			// send audit log message into provider tenant as user's tenant is null
-			this.auditLog.logConfigChange(action, cfgChange);
+		context.getCdsRuntime().requestContext().modifyUser(user -> user.setTenant(null)).run(requestContext -> {
+			context.getCdsRuntime().changeSetContext().run(changeSetContext -> {
```

Comment:
I don't understand this change. Why is a new ChangeSetContext here required? Should we also adapt the corresponding documentation: https://pages.github.tools.sap/cap/docs/java/auditlog/#subscriber-or-provider-tenant-in-multitenancy-scenario

Reply:
This is because I added now tenants to the mock users. The PersistenceService now complains that the tenant changed from "t1" to "null" during the transaction. This is not allowed.
Usually the Outbox detects that in case of MT + tenant "null" the in-memory outbox must be used. However this is not detected here, as the tests run in the single tenant scenario.

This creates a mismatch: PersistenceService behaves as if MT is enabled, while Outbox behaves as MT is disabled. This leads to a conflict, which requires this change to ensure that the log is written in its own transaction. That however goes against the intention of this code. There is no ideal solution for this at the moment.

The best possible change would be to make sure that the tenant is only added to the mock users in case of "local-mtxs". However this is currently also not easily possible, as mock users are maintained in a plain list.

Reply:
ok, thanks for clarification.